### ISSUE TITLE
Error adding a data file to a dataset

### ISSUE THREAD

**Issue by jeanetteclark** [NCEAS/metacatui#437]
Tried to add a new reformatted datafile with attributes and the following error popped up:

Error inserting or updating document: urn:uuid:9431dcb1-2bac-407c-a19d-68cad599ca45 since <?xml version="1.0"?><error>Error running xpath expression: //dateTimeDomain|//nonNumericDomain|//numericDomain|//access|//attributeList|//attribute|//constraint|//coverage|//temporalCoverage|//geographicCoverage|//taxonomicCoverage|//dataset|//dataSource|//dataTable|//otherEntity|//citation|//address|//conferenceLocation|//party|//originator|//creator|//contact|//publisher|//editor|//recipient|//performer|//institution|//metadataProvider|//associatedParty|//personnel|//physical|//connectionDefinition|//distribution|//researchProject|//project|//relatedProject|//software|//spatialRaster|//spatialReference|//spatialVector|//storedProcedure|//view|//protocol|//additionalMetadata : Error in xml document.  This EML document is not valid because the id urn-uuid-7fa804c2-e126-4e68-9c0c-40740175194d occurs more than once.  IDs must be unique.</error>

I am not sure what that pid is, I tried to read the sysmeta for it returned a no sysmeta found error.

This workflow generates the error:
Edit
Add file
Remove another file
Submit

**Comment by laurenwalker** [comment]
Was this in beta or test?

**Comment by jeanetteclark** [comment]
beta. All of these issues that I posted Friday happened in beta

**Comment by laurenwalker** [comment]
ok, thanks

**Comment by laurenwalker** [comment]
Fixed in 19a546d7bf1c5c963ef0833cf71e4590fb017855

I was able to reproduce this bug and determined that it was happening whenever I tried to replace a file in the dataset with a new file of the same name. Since we don't have a "Replace" feature  in the Editor yet, we are requiring users to simply upload the new data file and remove the old one. 

Whenever a file is added to the Editor, it will attempt to find the EMLEntity that belongs to that file/DataONEObject. In this case, an updated file is added to the package and paired up with it's existing `<otherEntity>` node based on file name. This happens during DataItemView.render(), since the "Describe" button is dependent on the EMLEntity validity/completeness. Shortly after, in DataPackage.handleAdd(), the package sees that a new file has been added and simple calls EML211.createEntity() to create a new EMLEntity model for the seemingly brand-new file.

The solution is basically to save a reference to the EMLEntity model in the DataONEObject model, and vice versa, so we can check if they have already been paired up. I also added some code to make sure that there is always a 1-to-1 relationship between EMLEntities and DataONEObjects.
